test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs

```diff
@@ -47,47 +49,85 @@ public void AddAspNetCoreInstrumentation_BadArgs()
         Assert.Throws<ArgumentNullException>(() => builder.AddAspNetCoreInstrumentation());
     }
 
-    [Fact]
-    public async Task RequestMetricIsCaptured()
+    [Theory]
+    [InlineData(null, true, false, 6, 0)] // emits old metric & attributes
+    [InlineData("http", false, true, 0, 5)] // emits new metric & attributes
+    [InlineData("http/dup", true, true, 6, 5)] // emits both old & new
+    public async Task RequestMetricIsCaptured(string environmentVarValue, bool validateOldSemConv, bool validateNewSemConv, int expectedOldTagsCount, int expectedNewTagsCount)
```

Comment:
I think it will be easier to do separate 3 tests. Will be easier to simply remove 2 of them later when we get rid of env variable.

Reply:
No strong opinion here. In previous PRs where I split tests out, I got pushback.

I do see that these tests are failing. I assume there's a concurrency issue with the Environment Variable.

Reply:
Its bit tricky to review the tests when they are combined in to one. When we remove them later it will require more work to readjust these tests rather than simply deleting the unwanted ones. @utpilla do you have any strong preference here?

>  I assume there's a concurrency issue with the Environment Variable.
Let's use IConfiguration.

Reply:
> Let's use IConfiguration.

100% agree. I looked into this earlier. The Options class has a ctor overload to take IConfiguration.
But I don't see an option to provide an instance of Options to the AddAspNetCoreInstrumentation() method.

Do you have any examples of how to do this?

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/459783f700b87325e057493d0715338d93e2de63/test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.server.cs#L126-L132

Reply:
> I think it will be easier to do separate 3 tests. Will be easier to simply remove 2 of them later when we get rid of env variable.

I agree with that. It would make it easier to remove the unwanted tests.

> No strong opinion here. In previous PRs where I split tests out, I got pushback.

@TimothyMothra Was that me? 😀. IIRC you had convinced me with the same rationale back then that it would make it easier to remove unwanted tests in future. We have dedicated unit tests for traces instrumentation, right? One for each possible value of the environment variable?

Reply:
I think it was you 😄 Yeah, we've been trying to separate tests methods. I can make that change here